### ISSUE TITLE
Mapcoloring example fails to work with windows

### ISSUE THREAD

**Issue by almullay** [dwavesystems/penaltymodel#46]
```
python dwave_mapcoloring.py
Traceback (most recent call last):
  File "dwave_mapcoloring.py", line 53, in <module>
    bqm = dwavebinarycsp.stitch(csp)
  File "C:\Python27\lib\site-packages\dwavebinarycsp\compilers\stitcher.py", line 164, in stitch
    pmodel = pm.get_penalty_model(spec)
  File "C:\Python27\lib\site-packages\penaltymodel\interface.py", line 59, in get_penalty_model
    pm = factory(specification)
  File "C:\Python27\lib\site-packages\penaltymodel_maxgap\interface.py", line 44, in get_penalty_model
    None)  # unspecified smt solver
  File "C:\Python27\lib\site-packages\penaltymodel_maxgap\generation.py", line 79, in generate_ising
    with Solver(smt_solver_name) as solver:
  File "C:\Python27\lib\site-packages\pysmt\shortcuts.py", line 817, in Solver
    **kwargs)
  File "C:\Python27\lib\site-packages\pysmt\factory.py", line 444, in Solver
    **options)
  File "C:\Python27\lib\site-packages\pysmt\factory.py", line 100, in get_solver
    **options)
  File "C:\Python27\lib\site-packages\pysmt\solvers\z3.py", line 146, in __init__
    self.z3 = z3.SolverFor(str(logic))
  File "C:\Python27\lib\site-packages\z3.py", line 6127, in SolverFor
    ctx = _get_ctx(ctx)
  File "C:\Python27\lib\site-packages\z3.py", line 210, in _get_ctx
    return main_ctx()
  File "C:\Python27\lib\site-packages\z3.py", line 205, in main_ctx
    _main_ctx = Context()
  File "C:\Python27\lib\site-packages\z3.py", line 151, in __init__
    conf = Z3_mk_config()
  File "C:\Python27\lib\site-packages\z3core.py", line 1215, in Z3_mk_config
    r = lib().Z3_mk_config()
  File "C:\Python27\lib\site-packages\z3core.py", line 24, in lib
    raise Z3Exception("init(Z3_LIBRARY_PATH) must be invoked before using Z3-python")
z3types.Z3Exception: init(Z3_LIBRARY_PATH) must be invoked before using Z3-python
Exception AttributeError: "Context instance has no attribute 'lib'" in <bound method Context.__del__ of <z3.Context instance at 0x000000000EA06DC8>> ignored
```

**Comment by almullay** [comment]
The code actually ran on Joe's Windows machine. He used python3 while I used 2.7.

**Comment by arcondello** [comment]
Addressed by #49
